PolyEngine/Games/SandboxRendering/Src/RenderingSandbox.cpp

```diff
@@ -100,38 +123,117 @@ void RenderingSandbox::Deinit()
+		DeferredTaskSystem::AddComponentImmediate<Poly::PointLightComponent>(world, PointLight, Color(1.0f, Random(), Random()), 2.0f, 0.1f);
+		Poly::TransformComponent* lightTrans = world->GetComponent<Poly::TransformComponent>(PointLight);
+		lightTrans->SetLocalTranslation(Pos + RndPos);
+	}
+}
+
+void GameMainSystem::GameUpdate(Poly::World* world)
+{
+	float time = (float)(world->GetWorldComponent<TimeWorldComponent>()->GetGameplayTime());
+	float deltaTime = (float)(TimeSystem::GetTimerDeltaTime(world, Poly::eEngineTimer::GAMEPLAY));
+
+	float di = 0;
+	for (auto cmpTuple : world->IterateComponents<DirectionalLightComponent, TransformComponent>())
+	{
+		TransformComponent* dirLightTrans = std::get<TransformComponent*>(cmpTuple);
+		Quaternion dirLightRot = Quaternion(Vector::UNIT_X, -45_deg);
+		dirLightRot *= Quaternion(Vector::UNIT_Y, 90.0_deg*di + 1_deg *100.0f*time);
+		
+		dirLightTrans->SetLocalRotation(dirLightRot);
+		// gConsole.LogInfo("DirLight[{}]: time: {}", di, time);
+
+		di += 1.0f;
+	}
+
+	int pi = 0;
+	for (auto cmpTuple : world->IterateComponents<PointLightComponent, TransformComponent>())
+	{
+		TransformComponent* pointLightTrans = std::get<TransformComponent*>(cmpTuple);
+		PointLightComponent* pointLight = std::get<PointLightComponent*>(cmpTuple);
+
+		// first point light stays in place
+		if (pi > 0)
+		{
+			float x = pointLightTrans->GetLocalTranslation().X;
+			pointLight->SetIntensity( 2.0f*Abs(Sin(1.0_rad *(x+time))) );
+		
+		}
+		else
+		{
+			Vector pointLightPos = Vector(0.0f, 2.0f, 0.0f);
+			// pointLightPos += Vector(Sin(1_rad*time), 0.0f, Cos(1_rad*time)) * 1.0f;
+			// pointLightPos += Vector(Cos(1_rad*time), 0.0f, 0.0f) * 10.0f;
+			// pointLightTrans->SetLocalTranslation(pointLightPos);
+		}
+			
+		// gConsole.LogInfo("PointLight[{}]: pos: {}", pi, pointLightTrans->GetLocalTranslation());
+		++pi;
+	}
+
+	// int UseCashetes = 0;
+	// float ColorTempLuminancePreservation = 0.75f;
+	// float Distortion	= 0.5f;
+	// float Bloom			= 0.4f;
+	// float ColorTempValue = 9500.0f; // 6500.0 from [1000.0, 40000.0]
+	// float Saturation	= 0.5f;
+	// float Grain			= 0.5f;
+	// float Stripes		= 0.5f;
+	// float Vignette		= 0.5f;
+	// float Brightness	= 0.1f;
+	// float Contrast		= 1.5f;
+
+	// for (auto cmpTuple : world->IterateComponents<PostprocessSettingsComponent>())
+	// {
+	// 	PostprocessSettingsComponent* postCmp = std::get<PostprocessSettingsComponent*>(cmpTuple);
+	// 	float t = SmoothStep(-0.2f, 0.2f, -Cos(Angle::FromRadians(0.5f*time))); // [0, 1]
+	// 	float t = 0.0f;
+	// 
+	// 	postCmp->Distortion = t * Distortion;
+	// 	postCmp->ColorTempValue = Lerp(6500.0f, ColorTempValue, t);
+	// 	postCmp->Saturation = Lerp(1.0f, Saturation, t);
+	// 	postCmp->Grain = Lerp(0.0f, Grain, t);
+	// 	postCmp->Stripes = Lerp(0.0f, Stripes, t);
+	// 	postCmp->Vignette = Lerp(0.0f, Vignette, t);
+	// 
+	// 	gConsole.LogInfo("PostCmp: t: {}, Saturation: {}, Vignette: {}", t, postCmp->Saturation, postCmp->Vignette);
+	// }
```

Comment:
Commented blocks of code should never be pushed to repository.

Reply:
Sorry